src/cascadia/TerminalApp/TabManagement.cpp

```diff
@@ -362,25 +364,38 @@ namespace winrt::TerminalApp::implementation
         {
             if (const auto control{ tab.GetActiveTerminalControl() })
             {
-                // GH#11356 - we can't use the UWP apis for writing the file,
-                // because they don't work elevated (shocker) So just use the
-                // shell32 file picker manually.
-                auto path = co_await SaveFilePicker(*_hostingHwnd, [control](auto&& dialog) {
-                    THROW_IF_FAILED(dialog->SetClientGuid(clientGuidExportFile));
-                    try
-                    {
-                        // Default to the Downloads folder
-                        auto folderShellItem{ winrt::capture<IShellItem>(&SHGetKnownFolderItem, FOLDERID_Downloads, KF_FLAG_DEFAULT, nullptr) };
-                        dialog->SetDefaultFolder(folderShellItem.get());
-                    }
-                    CATCH_LOG(); // non-fatal
-                    THROW_IF_FAILED(dialog->SetFileTypes(ARRAYSIZE(supportedFileTypes), supportedFileTypes));
-                    THROW_IF_FAILED(dialog->SetFileTypeIndex(1)); // the array is 1-indexed
-                    THROW_IF_FAILED(dialog->SetDefaultExtension(L"txt"));
-
-                    // Default to using the tab title as the file name
-                    THROW_IF_FAILED(dialog->SetFileName((control.Title() + L".txt").c_str()));
-                });
+                auto path = filepath;
+
+                if (path.empty())
+                {
+                    // GH#11356 - we can't use the UWP apis for writing the file,
+                    // because they don't work elevated (shocker) So just use the
+                    // shell32 file picker manually.
+                    path = co_await SaveFilePicker(*_hostingHwnd, [control](auto&& dialog) {
+                        THROW_IF_FAILED(dialog->SetClientGuid(clientGuidExportFile));
+                        try
+                        {
+                            // Default to the Downloads folder
+                            auto folderShellItem{ winrt::capture<IShellItem>(&SHGetKnownFolderItem, FOLDERID_Downloads, KF_FLAG_DEFAULT, nullptr) };
+                            dialog->SetDefaultFolder(folderShellItem.get());
+                        }
+                        CATCH_LOG(); // non-fatal
+                        THROW_IF_FAILED(dialog->SetFileTypes(ARRAYSIZE(supportedFileTypes), supportedFileTypes));
+                        THROW_IF_FAILED(dialog->SetFileTypeIndex(1)); // the array is 1-indexed
+                        THROW_IF_FAILED(dialog->SetDefaultExtension(L"txt"));
+
+                        // Default to using the tab title as the file name
+                        THROW_IF_FAILED(dialog->SetFileName((control.Title() + L".txt").c_str()));
+                    });
+                }
+                else
+                {
+                    // The file picker isn't going to give us paths with

```

Comment:
Honestly wouldn't hurt you to call this on either path just in case, but it's fine.